src/SonarQube.Client/Services/RequestFactory.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Client
+{
+    public class RequestFactory
+    {
+        private readonly Dictionary<Type, SortedList<Version, Func<object>>> requestMappings =
+            new Dictionary<Type, SortedList<Version, Func<object>>>();
```

Comment:
Does this have to be object, or could it be IRequest?

Reply:
It could be IRequest